pkg/agent/discoverycache.go

```diff
@@ -105,6 +110,9 @@ func (j *discoveryCache) updateAPICache() {
 	)
 
 	for _, svc := range apiServices {
+		if j.migrator != nil {
+			svc, _ = j.migrator.Migrate(svc)
```

Comment:
I found leaving the migrator in the discovery cache to be the most efficient way to fetch a resource, update it, then save it to the cache. @vivekschauhan Let me know if you are ok with this, or if you have another idea based on how you want to populate the initial cache.

Reply:
Looks fine except iterating over the cached APIService(from persisted cache if available) in streaming mode